### PR TITLE
Support DENSE column in codegen.go

### LINKED ISSUE #546: Support DENSE column in codegen.go
**Is your feature request related to a problem? Please describe.**
For input table like 

<table>
  <tr>
    <th colspan="2">my_table</th>
  </tr>
  <tr>
    <td>dense</td>
    <td>class</td>
  </tr>
  <tr>
    <td>"1.2,1.7,0.6"</td>
    <td>0</td>
  </tr>
  <tr>
    <td>"0.2,1.7,0.6"</td>
    <td>1</td>
  </tr>
  <tr>
    <td>"0.2,-1.6,0.6"</td>
    <td>0</td>
  </tr>
  <tr>
    <td>...</td>
    <td>...</td>
  </tr>
</table>

We should be able to

```SQL
SELECT * FROM my_table
TRAIN DNNClassifier
COLUMN DENSE(dense, 3, comma)
LABEL class
INTO my_model_table
```

**Describe the solution you'd like**
Add support for `DENSE` column in `codegen.go`.

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Fix #546.

### CODE REVIEW

File: sql/codegen.go [hunk 1]
```diff
@@ -134,6 +134,28 @@ func newFiller(pr *extendedSelect, fts fieldTypes, db *DB) (*filler, error) {
 				r.FeatureColumnsCode[target],
 				feaColCode)
 		}
+		for _, spec := range colSpecs {
```
Comment (typhoonzero) [inline sql/codegen.go]: One thing to discuss is, whether we need to involve `featureSpec` in `codegen.go`, in `codegen_alps.go` `featureSpec` and `featureColumn` are both needed to:

- `featureSpec`: Used to generate code for `input_fn` and dataset
- `featureColumn`: Used to generate `feature_column` code to pass to `Estimators`

for one column, `DENSE(col)` and `NUMERIC(col)` must both appear to generate correct code, and it's a bit hard to use.

So I'd like to consider use only one structure to store all column's meta information.

Comment (tonyyang-svail) [reply sql/codegen.go]: > So I'd like to consider using only one structure to store all column's meta information.

Sure, I will try to implement it as `NUMERIC(field_name, 4)`.

Comment (tonyyang-svail) [reply sql/codegen.go]: Done.

File: sql/executor_test.go [hunk 2]
```diff
@@ -80,6 +81,32 @@ func TestExecutorTrainAndPredictDNNLocalFS(t *testing.T) {
...
+	a := assert.New(t)
+	a.NotPanics(func() {
+		stream := Run(`SELECT * FROM iris.train_dense
+TRAIN DNNClassifier
+WITH
+n_classes = 3,
+hidden_units = [10, 20],
+EPOCHS = 200,
+BATCHSIZE = 10
+COLUMN NUMERIC(dense, 4)
```
Comment (typhoonzero) [inline sql/executor_test.go:97]: maybe use data `text_cn.train_processed` to test dense column with delimiter?

Comment (tonyyang-svail) [reply sql/executor_test.go:97]: I think one advantage of using `iris.train_dense` is we are able to check the training accuracy.

Comment (typhoonzero) [reply sql/executor_test.go:97]: I mean, both tests are needed to cover dense column with a delimiter and without should be helpful. Thanks!!

**Comment by typhoonzero** [comment]
Please fix the CI tests after merging so we can merge this PR~

**Review by typhoonzero — APPROVED** [review]
LGTM, TODO: add docs to descirbe column clause usages.

**Pull request merged**